src/WebSocketHandler.js

```diff
@@ -12,28 +12,37 @@ import WS from './websocket';
 import EventEmitter from 'events';
 
 class WebSocketHandler extends EventEmitter {
+  constructor() {
+    super();
+
+    this.ws = null;
+  }
+
   setup() {
-    this.ws = new WS({ wsURL: helpers.getWSServerURL });
-
-    this.on('is_online', (value) => {
-      if (value) {
-        wallet.onWebsocketOpened();
-        this.ws.emit('reload_data');
-      } else {
-        wallet.onWebsocketBeforeClose();
+    if (this.ws === null) {
+      this.ws = new WS({ wsURL: helpers.getWSServerURL });
+
+      this.on('is_online', (value) => {
+        if (value) {
+          wallet.onWebsocketOpened();
+          this.ws.emit('reload_data');
+        } else {
+          wallet.onWebsocketBeforeClose();
+        }
+      });
+
+      /*
+       * This class still exists for compatibility reasons
+       * it is used in some wallets and in our old lib code
+       * In the wallets we capture some events from it, so
+       * this following code is to emit all events emitted from this.ws
+       */
+      this.oldEmit = this.ws.emit;
+      this.ws.emit = (type, data) => {
+        this.emit(type, data);
+        console.log('Web Socket Handler', type, data);
+        return this.oldEmit(type, data);
       }
-    });
-
-    /*
-     * This class still exists for compatibility reasons
-     * it is used in some wallets and in our old lib code
-     * In the wallets we capture some events from it, so
-     * this following code is to emit all events emitted from this.ws
-     */
-    this.oldEmit = this.ws.emit;
-    this.ws.emit = (type, ...args) => {
-      this.emit(type, ...args);
-      return this.oldEmit(type, ...args);
     }
 
     // To keep compatibility with methods previously used in this singleton
```

Comment:
Do we need to execute this even if `this.ws` was not null? I imagine the setup would be done already.

Reply:
It's important to do the `setup` again. Look at the `endConnection` code.

```
  endConnection() {
    this.setIsOnline(undefined);
    this.started = false;
    this.connected = undefined;
    if (this.ws) {
      this.ws.onclose = () => {};
      this.ws.close();
      this.ws = null;
    }
    clearInterval(this.heartbeat);
  }
```

If you do an `endConnection` in the `src/websocket.js` file and then `setup`, `this.ws` won't be `null` but we would need to execute the setup all over again.

However, I think I will follow @msbrogli suggestion in the other comment and set `this.ws = null` when ending the connection, so this should never happen.

I believe it doesn't hurt to keep like this only to protect in case this happen again. This `WebSocketHandler` is the file that will be removed in the next lib refactor.

Reply:
Ok